src/Compilers/CSharp/Test/Syntax/Parsing/PatternParsingTests.cs

```diff
@@ -11294,5 +11294,100 @@ public void GenericTypeAsTypePatternInSwitchExpression()
             }
             EOF();
         }
+
+        [Fact, WorkItem(48112, "https://github.com/dotnet/roslyn/issues/48112")]
+        public void NullableTypeAsTypePatternInSwitchExpression()
+        {
+            UsingStatement(@"_ = e switch { int? => 1 };",
```

Comment:
can we have some explicit tests of `foo is int ? x : y` to ensure these don't regress.  thanks!

Reply:
`IsNullableType01` and `IsNullableType02` already do that for `o is A ? b : c` and `o is A? ? b : c`

Is there any other scenario worth adding test?

Reply:
Note this doesn't affect `is`. Actually `T?` is allowed in `is` for historical reasons. Semantically it's identical to `T`, but in patterns we have an explicit diagnostic on `Nullable<T>`. This fix reports the same error on `T?` syntax.

Reply:
@gafter  would be a good person to weigh in on where we should focus tests.  Even if this isn't directly impacted, those tests will help prevent regressions as this code gets refactored in the future.

Reply:
I'm actually not sure the compiler as it is today complies with the spec here. `switch` expression patterns can be a `constant_pattern`, and `constant_pattern` is just a `constant_expression`. That means we should accept code of this form: `o switch { S ? true : false => false  }` (when `S` is a constant), but we do not today. Do we have tests for similar scenarios in switch statements (which do accept this code)? For example, this code:
```cs
    switch (o)
    {
        case S ? true : false: // Totally legal if S is constant
            break;
        case int?: // Error: ':' expected
            break;
    }
```
While I agree with this in principle, we definitely need to have a lot of tests.

Reply:
we do accept conditionals but only when parenthesized.

Reply:
> we do accept conditionals but only when parenthesized.

Yes, that is the current behavior. It's not to spec 🙂. I'm uncomfortable making this change without the behavior around ternaries actually being correct.

Reply:
To expand on my concern: The first thing that we'd need to do to correctly implement the spec is revert this change. I think that if we wanted to limit the error recovery to _just_ scenarios that can result from a bad pattern, such as `int? =>`, I would be more comfortable with that. So, if the character after the `?` is a `=`, it seems reasonable to recover there. It also seems like it would be reasonable to recover after `int?:` in a case label: today, the error you get for `case int?:` is `:` expected, which is just confusing.

Reply:
:memo: The cases you mentioned happen only in the top level patterns - if we're going to look only for those two characters to disambiguate, we need to make sure it won't break when these appear deep nested in recursive patterns.

btw, is this an outstanding issue or was there an actual reason or parsing difficulties that we didn't cover these the first time?

Reply:
> btw, is this an outstanding issue or was there an actual reason or parsing difficulties that we didn't cover these the first time?

I suspect it's just a bug.

Reply:
One last note. If we fix the conditional parsing, we're practically breaking those using an older compiler to compile a program written in a newer version but both with the same langversion. I guess it should be fine considering this is just an edge case unlikely to actually happen.

Reply:
Yeah, I wouldn't be worried about that. It's a pretty edge of edge case.